midi_app_controller/gui.py

```diff
@@ -0,0 +1,426 @@
+import sys
+from pathlib import Path
+from app_model.types import Action
+import yaml
+from PyQt5.QtWidgets import QApplication, QWidget, QVBoxLayout, QPushButton, QLabel, QMainWindow, QAction, QHBoxLayout, \
+    QDial, QComboBox, QFormLayout, QMenuBar
+from midi_app_controller.actions.bound_controller import BoundController, ButtonActions, KnobActions
+from midi_app_controller.models.binds import ButtonBind, KnobBind, Binds
+
+
+class MIDIControllerView(QMainWindow):
+    def __init__(self):
+        super().__init__()
+        self.setupUI()
+        self.loadSettings()
+
+    def ui_action_to_model_action(self, action_name: str):
+        """
+        Converts an action name to a model Action.
+
+        Parameters
+        ----------
+        action_name: str
+            Name of the action.
+
+        Returns
+        -------
+        Action
+            An instance of Action with the given name.
+        """
+
+        if action_name == 'None':
+            return None
+        return Action(id=action_name, callback="", title=action_name)
+
+    def setupUI(self):
+        """
+        Sets up the UI.
+        """
+
+        self.createCentralWidget()
+        self.createMenuBar()
+        self.layoutControls()
+        self.setWindowTitle('MIDI Controller')
+
+    def createCentralWidget(self):
+        """
+        Creates the central widget.
+        """
+
+        self.central_widget = QWidget()
+        self.setCentralWidget(self.central_widget)
+        self.main_layout = QVBoxLayout(self.central_widget)
+        self.setStyleSheet("background-color: #2e2e2e;")
+
+    def createMenuBar(self):
+        """
+        Creates the menu bar.
+        """
+
+        menubar = self.menuBar()
+        self.addMenuAction(menubar, 'Save Settings', self.saveSettings)
+        self.addMenuAction(menubar, 'Load Settings', self.loadSettings)
+
+    def addMenuAction(self, menu: QMenuBar, title: str, method: callable):
+        """
+        Adds an action to the menu.
+
+        Parameters
+        ----------
+        menu: QMenuBar
+            Menu to add the action to.
+        title: str
+            Title of the action.
+        method: callable
+            Method to be called when the action is triggered.
+        """
+
+        action = QAction(title, self)
+        action.triggered.connect(method)
+        menu.addAction(action)
+
+    def layoutControls(self):
```

Comment:
Controller elements should be loaded from a config file based on the `Controller` class.